packages/daemon/src/app/controller/job.ts

```diff
@@ -17,25 +19,51 @@ export class JobController extends BaseEventController {
   @inject('pluginManager')
   pluginManager: PluginManager;
 
+  private async setupTracer(job: JobEntity) {
+    const logPath = join(constants.PIPCOOK_RUN, job.id, 'logs');
+    const stdoutFile = join(logPath, 'stdout.log');
+    const stderrFile = join(logPath, 'stderr.log');
+    await ensureDir(logPath);
+    await Promise.all([
+      ensureFile(stdoutFile),
+      ensureFile(stderrFile)
+    ]);
+    const tracer = await this.traceManager.create({ stdoutFile, stderrFile });
+    return tracer;
+  }
+
+  private createParams(pipeline: PipelineEntity, jobParams?: JobParam[]): JobParam[] {
+    const params: JobParam[] = [];
+
+    for (const pluginType of constants.PLUGINS) {
+      const tempParam = JSON.parse(pipeline[`${pluginType}Params`]);
+      let jobParam = [];
+      if (jobParams) {
+        jobParam = jobParams
+          .filter((it) => it.pluginType === pluginType)
+          .map((it) => it.data);
+      }
+
+      params.push({ pluginType, data: Object.assign(tempParam, ...jobParam) });
+    }
+    return params;
+  }
+
   /**
    * start a job from pipeline id or name
    */
   @post()
   public async run(): Promise<void> {
     const { pipelineId } = this.ctx.request.body;
+    const { params : updateParams = [] } = this.ctx.request.body;
```

Comment:
As we have defined a GET method `/:id/parameters`, thus we need to use that on the request body?